apps/cowswap-frontend/src/modules/volumeFee/state/volumeFeeAtom.ts

```diff
@@ -7,9 +7,25 @@ import { injectedWidgetPartnerFeeAtom } from 'modules/injectedWidget'
 import { tradeTypeAtom } from 'modules/trade'
 import { TradeType } from 'modules/trade/types/TradeType'
 
+import { featureFlagsAtom } from 'common/state/featureFlagsState'
+
+import { cowSwapFeeAtom } from './cowswapFeeAtom'
+
 import { VolumeFee } from '../types'
 
 export const volumeFeeAtom = atom<VolumeFee | undefined>((get) => {
+  const featureFlags = get(featureFlagsAtom)
+  const cowSwapFee = get(cowSwapFeeAtom)
+  const widgetPartnerFee = get(widgetPartnerFeeAtom)
+
+  if (featureFlags.isCowSwapFeeEnabled) {
+    return cowSwapFee || widgetPartnerFee
+  }
```

Comment:
Why return the `widgetPartnerFee` when `cowSwapFee` isn't defined?
If cowswap fee is enabled only that should be returned, no?

Reply:
If `cowSwapFee` is not defined it might be a case when we in the widget mode, hence we should use widget partner fee